server/src/test/java/org/elasticsearch/index/query/MatchQueryBuilderTests.java

```diff
@@ -258,12 +258,33 @@ public void testSimpleMatchQuery() throws IOException {
               }
             }""";
         MatchQueryBuilder qb = (MatchQueryBuilder) parseQuery(json);
-        checkGeneratedJson(json, qb);
+        checkGeneratedJson("""
```

Comment:
Could you add a comment (either here or javadoc on the test) explaining that the generated JSON is different than the "source" is because default fields are dropped? It looks like this is the only test here that's checking that.